app/src/androidTest/java/org/mozilla/fenix/perf/HomeFragmentBenchmark.kt

```diff
@@ -0,0 +1,129 @@
+    )
+
+    @get: Rule
+    val intentReceiverActivityTestRule = ActivityTestRule(
+        IntentReceiverActivity::class.java, true, false
+    )
+
+    @Before
+    fun setup() {
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+
+        genericURL = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        intent = Intent(Intent.ACTION_VIEW, genericURL.url)
+        intentReceiverActivityTestRule.launchActivity(intent)
+    }
+
+    /**
+     * Benchmarks the time taken from clicking the "Home Button" on the URL bar located on the
+     * BrowserFragment up until the HomeFragment is shown and usable.
+     *
+     * NOTE: According to Google's documentations, this test should be annotated with @UiThreadTest.
+     * However, doing so will make UiAutomator or Espresso API's unusable since the test will hang.
+     * Instead, we use the `runOnUiThread{}` which is the method that @UiThreadTest uses.
+     */
+    @Test
+    fun homeScreenBenchmark() {
+        var homeButton: View? = null
+
+        benchmarkRule.measureRepeated {
+            runWithTimingDisabled {
+                homeButton = getHomeButton(homeActivityRule.activity.findViewById<BrowserToolbar>(R.id.toolbar))
+            }
+            homeActivityRule.runOnUiThread {
+                homeButton!!.callOnClick()
+            }
+
+            // The homescreen could potentially take a while to show on the screen.
+            // Therefore, we should wait and make sure that the Main Thread is idled.
+            // `waitForIdleSync()` waits for the message queue to be empty. Source:
+            // https://android.googlesource.com/platform/frameworks/base/+/8f6f1f43eeb0d7263b626978ee2c21d4053bf610/core/java/android/app/Instrumentation.java#328
+            InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+
+            runWithTimingDisabled {
+                // Make sure that everything is shown on screen. The recyclerview should be the last
+                // thing to be drawn.
+                assertNotNull(mDevice.findObject(By.res(TestHelper.packageName, resourceId)))
+
+                homeButton = null
+
+                // Click on the "Jump Back In" tab that was created  in the `Setup()`. This should be
+                // the third item in the recyclerview.
+                onView(withId(R.id.sessionControlRecyclerView)).perform(
+                    RecyclerViewActions.actionOnItemAtPosition<ViewHolder>(2, click())
+                )
+            }
+        }
+    }
+
+    /**
+     * DFS search for the home screen button since it is created dynamically in
+     * `BaseBrowserFragment.kt`. This AppCompatButton is created as an action and doesn't have a
+     * resource ID making it hard to do `findViewById`. However, since it contains a contentDescription
+     * we can find it with it.
+     */
+    private fun getHomeButton(view: View): View? {
```

Comment:
suggestion (non-block): I found this [`View.findViewsWithText`](https://developer.android.com/reference/android/view/View#findViewsWithText(java.util.ArrayList%3Candroid.view.View%3E,%20java.lang.CharSequence,%20int)) method that operates on content descriptions (if you pass in the right flag). I wonder if we should replace `getHomeButton` with this method. I'm concerned about people copy-pasting this test so we have many duplicate DFS methods (which may accrue bugs while pasting) when we could use the built-in method instead.